src/pages/groupComparison/UpSet.tsx

```diff
@@ -0,0 +1,492 @@
+        } else {
+            categoryDomain = [0, 0];
+        }
+        return { x: categoryDomain, y: countDomain };
+    }
+
+    @computed get scatterPlotDomain() {
+        let countDomain: number[];
+        let categoryDomain: number[];
+        if (this.data.length > 0) {
+            categoryDomain = [this.categoryCoord(0), this.categoryCoord(Math.max(1, this.data.length - 1))];
+            countDomain = [this.categoryCoord(0), this.categoryCoord(Math.max(1, _.keys(this.groupLabels).length - 1))]
+        } else {
+            categoryDomain = [0, 0];
+            countDomain = [0, 0];
+        }
+        return { x: categoryDomain, y: countDomain };
+    }
+
+    @computed get categoryTickValues() {
+        return this.data.map((x, i) => this.categoryCoord(i));
+    }
+
+    @computed get groupTickValues() {
+        return _.map(this.groupLabels, (label, index) => this.categoryCoord(index));
+    }
+
+    @autobind
+    private formatNumericalTick(t: number, i: number, ticks: number[]) {
+        return tickFormatNumeral(t, ticks);
+    }
+
+    @autobind
+    private groupTick(t: number, index: number) {
+        return this.groupLabels[index];
+    }
+
+    @computed get scatterPlotTopPadding() {
+        // subtract 100 to plot scatter p\lot close to bar plot
+        return this.barChartHeight - 100;
+    }
+
+    private tooltip(datum: any) {
+        const groupNames = _.map(datum.groups as string[], groupUid => this.uidToGroup[groupUid]);
+        return (
+            <div>
+                <strong>{datum.cases.length} case{datum.cases.length > 0 ? 's' : ''}</strong><br />
+                {groupNames.length > 1 ? 'common ' : ''}in {joinNames(groupNames, "and")}
+            </div>
+        );
+    }
+
+    @autobind
+    private getTooltip() {
+        if (this.container && this.tooltipModel) {
+            const countAxisOffset = (this.tooltipModel.y + this.tooltipModel.y0) / 2;
+            const categoryAxisOffset = this.tooltipModel.x;
+            return (
+                <ScatterPlotTooltip
+                    placement={"right"}
+                    container={this.container}
+                    targetHovered={this.pointHovered}
+                    targetCoords={{
+                        x: categoryAxisOffset + this.leftPadding,
+                        y: countAxisOffset + this.topPadding
+                    }}
+                    overlay={this.tooltip(this.tooltipModel.datum)}
+                    arrowOffsetTop={20}
+                />
+            );
+        } else {
+            return <span></span>;
+        }
+    }
+
+    @computed private get getChart() {
+        if (this.data.length > 0) {
+            return (
+                <div
+                    ref={this.containerRef}
+                    style={{ width: this.svgWidth, height: this.svgHeight }}
+                >
+                    <svg
```

Comment:
what's reason for custom SVG wrapper for VictoryChart?

Reply:
actually I used different `<VictoryChart>`  for each bar and scatter plot. to merge them I used svg